spring/spring4/spring4-common/src/main/java/org/infinispan/spring/session/AbstractInfinispanSessionRepository.java

```diff
@@ -0,0 +1,62 @@
+package org.infinispan.spring.session;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.spring.provider.SpringCache;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.session.MapSession;
+import org.springframework.session.SessionRepository;
+
+/**
+ * Infinispan implementation for Spring Session with basic functionality.
+ *
+ * @author Sebastian Łaskawiec
+ * @see <a href="http://projects.spring.io/spring-session">Spring Session Web Page</a>
+ * @see SessionRepository
+ * @see ApplicationEventPublisherAware
+ * @since 9.0
+ */
+public abstract class AbstractInfinispanSessionRepository implements SessionRepository<MapSession>, ApplicationEventPublisherAware {
+
+   protected final AbstractApplicationPublisherBridge applicationEventPublisher;
+   protected final SpringCache cache;
+   protected final PrincipalNameResolver principalNameResolver = new PrincipalNameResolver();
+
+   protected AbstractInfinispanSessionRepository(SpringCache cache, AbstractApplicationPublisherBridge eventsBridge) {
+      Objects.requireNonNull(cache, "SpringCache can not be null");
+      Objects.requireNonNull(eventsBridge, "EventBridge can not be null");
+      applicationEventPublisher = eventsBridge;
+      this.cache = cache;
+   }
+
+   @Override
+   public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+      this.applicationEventPublisher.setApplicationEventPublisher(applicationEventPublisher);
+   }
+
+   @Override
+   public MapSession createSession() {
+      MapSession result = new MapSession();
+      result.setCreationTime(System.currentTimeMillis());
+      return result;
+   }
+
+   @Override
+   public void save(MapSession session) {
+      cache.put(session.getId(), session, session.getMaxInactiveIntervalInSeconds(), TimeUnit.SECONDS);
```

Comment:
It looks you are right... when I invoke `Cache#put(K, V, lifespan, unit)` method, we set a "create time" for an entry in `DataContainer`. The `EvictionManager` is interested only in comparing this "create time" against the wall clock. So even very heavily accessed session will be evicted after certain amount of time...

@wburns May I ask you for opinion? Is there any way to "refresh" an entry when accessing it? So it won't expire?

@mgencur Answering your question about maxIdle - by default it is set to [maximum value](https://github.com/infinispan/infinispan/pull/4562/files#diff-e1c0e3c22577a8e634f5e7d0578ede8fR52) in [here](https://github.com/infinispan/infinispan/pull/4562/files#diff-8cee8963429ccfba417a96e6ac5f93a4R32). I decided to make it configurable at annotation level since all Spring Session implementations provide this kind of feature.